app-marketplace-v2-spec/architecture.md

```diff
@@ -18,15 +18,19 @@
 
 ### Overview.
 1. Mattermost interacts with **Cloud Apps** in 2 ways: **Commands** that have request/response semantics, and 1-way **Event notifications**.
+  - Dani Q: Alternatively, **Event notifications** can be sent to the clients, and the clients trigger commands if needed.
```

Comment:
Maybe we should have both. My idea is that if the "service provider" (plugin or external) stored no information, they would have no use for the events. But the clients could just perform the needed action. i.e.:
- If we send an event to the service provider, the provider must know all the clients connected. And then it has to send them some information.
- If we send an event to the active clients, they just have to execute the related action, and only active clients will perform it.

For example, an affinity plugin, telling you which person on this channel has the most affinity to you (because they are in the same channels).
- If we send the "user joined channel" event to the plugin, the plugin must know which clients are active to update the information they are seeing. Then it has to send all the clients the new information, for which the plugin has to configure its own websockets.
- If we send the event to the clients, the clients just need to call the plugin and say "hey, recalculate my affinity", and they can update their state without the plugin store any extra information.